types/game.d.ts

```diff
@@ -4,7 +4,7 @@
  * - `/stream`: as `{ ready: false }`
  * @defaultValue `null`
  */
-declare let canvas: Canvas | { ready: boolean } | null;
+declare let canvas: Canvas | null;
```

Comment:
The reason that `{ ready: boolean }` is in the union is because in the stream view the `canvas`  is just `{ ready: false }`. I'm honestly torn on this part. I think that as long as the null is part of the union, that should be as well. Making it `declare let canvas: Canvas | { ready: false } | null;` would be best for now as that should let you guard it by doing `if (!canvas?.ready) return;` in functions where it's used.

Reply:
You can also just guard it by
```ts
if(!canvas instanceof Canvas) return;
```
But I agree. We should stay true to what it is actually being set to, i.e.
```ts
declare let canvas: Canvas | { ready: false } | null;
```

Reply:
Alright, added `{ ready: false }` to the canvas type as it could explicitly be that type of object, no matter how counter-intuitive that is. The source is what it is :)

Reply:
Looks like my method of typing this doesn't need to change, either.
```typescript
      if (canvas !== null && canvas.ready) {
        // do stuff
      }
```